### ISSUE TITLE
meaning of 'alpha' in the SE kernel

### ISSUE THREAD

**Issue by gaetanoscandariato** [nespinoza/juliet#118]
Hi Nestor,

I'm playing aroud with the GP functionality in juliet and I have some difficulties in understanding the alpha parameter passed to george.kernels (v 0.4.0). In the doc of george I see that the input parameter for the ExpSquaredKernel is the metric. In your fit.py, line 4334, you set the kernel parameter to np.log( 1/alpha ). What is the reason for this definition of alpha?

Thanks in advance

**Comment by nespinoza** [comment]
Hi @gaetanoscandariato,

 The `george` documentation [here](https://george.readthedocs.io/en/latest/user/kernels/) states that "(...) the parameter vector is the logarithm of the parameters given when building the kernel. This will be the case for any strictly positive parameters because it is always better to fit in the logarithm of these types of parameters (...)". That's the reason for the logarithm in the code when using the `set_parameter_vector` for GP hyperparameters (which the user doesn't see).

 As for the 1 / alpha --- the reason is that the metrics in `george` divide the terms inside the functions (see, e.g., https://george.readthedocs.io/en/latest/tutorials/hyper/). That is, when you set a parameter `x` via the `set_parameter_vector` for, say, the ExpSquaredKernel, you are actually defining`exp(- r^2 / 2 x (x))`. By defining `1 / alpha`, you are defining the inverse length-scale. This is useful because as it goes to zero, the kernel dies off too.
 
  So, bottom line, `alpha` is an inverse length-scale --- as [defined in the `juliet` tutorial](https://juliet.readthedocs.io/en/latest/user/priorsnparameters.html). Let me know if something is not clear (or if you spot incongruences above!).

 Thanks!

N.

**Comment by gaetanoscandariato** [comment]
ok, that makes sense.

I got tricked by a publication where alpha is days^-2, probably an old implemetation of juliet? Anyway, never mind.

Thanks again

G

**Comment by nespinoza** [comment]
Hi @gaetanoscandariato, 

 Actually, I messed up the comment above. It is, indeed, the inverse *squared* length-scale for the squared exponential kernel (edited my statement above). To see this, note the [following note](https://george.readthedocs.io/en/latest/user/kernels/) in the `george` documentation:

<img width="740" alt="Screen Shot 2024-05-24 at 10 20 58 AM" src="https://github.com/nespinoza/juliet/assets/2046682/8c7d9b8e-ae1f-4a05-a570-a45666f5d281">

 The metric, as defined above, is dividing the distance _squared_. So the metric is the squared length-scale. Because in `juliet`, `alpha = 1 / lambda` in the notation above, `alpha` is indeed the inverse length-scale squared. I think [the `george` tutorial](https://george.readthedocs.io/en/latest/tutorials/hyper/) also clarifies this in an example; note that in that tutorial, the kernel `k1` is defined as:

`k1 = 66**2 * kernels.ExpSquaredKernel(metric=67**2)`

 Here, `67` (ppm) is the actual length-scale of the process, but it has to be squared to match the `george` definition.[ In the `juliet` explainer of the priors](https://juliet.readthedocs.io/en/latest/user/priorsnparameters.html), we also note this: "Inverse (squared) length-scale/normalized (...)".

 Hope this helps clarify things!

Néstor

**Comment by gaetanoscandariato** [comment]
Hi @nespinoza ,

thanks for the clarification. Now I accept the squaring of the lengthscale to get the metric for george, and also get the meaning of alpha in juliet.

There are though some subtleties in my data analysis that led me to a possible glitch inside juliet. In your previous message you say that juliet needs alpha as the squared inverse of the lengthscale, but in `gaussian_process.set_parameter_vector()`, inside `elif self.kernel_name == 'ExpSineSquaredSEKernel':`, the metric parameter is passed as np.log(1/alpha**2). The log is also applied to sigma, but I haven't investigated yet if this is correct or not.

Can you please give me feedback on this?

Thanks again

**Comment by nespinoza** [comment]
Hi @gaetanoscandariato --- _excellent_ catch. I think you are correct; we are setting the parameter vector in linear, but that is setting the _log_ `Gamma`, not the `Gamma`. I'll run some tests and fix this in `juliet` if this is indeed the case (I think it is, from some initial tests).

N.

**Comment by gaetanoscandariato** [comment]
ok, just please let me know because I have edited your source code in many parts and I can't redo everything from scratch with an updated version of juliet. It would be easier for me to fix this bug in my local version. thanks :)

**Comment by nespinoza** [comment]
I have upgraded `juliet` with other things including this one --- the newest `2.2.6` is out in both `pip` and `master` (https://github.com/nespinoza/juliet). Closing this as solved!

N.

**Comment by ParcLena** [comment]
Hi Nestor,

I was going through the issues when I saw this one, which interests me because I use the ExpSquaredKernel for my fits. I thought it would be a good idea to update my version of juliet (I was in 2.2.1). So I updated juliet (to 2.2.7 then 2.2.6) and tried to make a simple GP SE fit on a set of RVs FWHM (which worked with 2.2.1 and up to 2.2.5). Except that the fit started to diverge greatly. So I don't know if I need to change something in my priors or if something has changed that has totally broken my code with this version? 

Léna
